hostmetricreceiver: cannot be compiled for operating systems other than Linux

### Component(s)

receiver/hostmetrics

### What happened?

## Description

hostmetricreceiver cannot be compiled for operating systems other than Linux

## Steps to Reproduce

## Expected Result

Successful build

## Actual Result

```bash
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper
# github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper
internal/scraper/processscraper/process_scraper_others.go:21:9: cannot use nil as string value in return statement
internal/scraper/processscraper/process_scraper_others.go:25:9: cannot use nil as string value in return statement
```


### Collector version

v0.81.0

### Environment information

## Environment
OS: Fedora 38
Compiler(if manually compiled): go1.21rc2


### OpenTelemetry Collector configuration

```yaml
None
```


### Log output

_No response_

### Additional context

_No response_

--- a/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_others.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_others.go
@@ -18,11 +18,11 @@ func (s *scraper) recordCPUTimeMetric(now pcommon.Timestamp, cpuTime *cpu.TimesS
 func (s *scraper) recordCPUUtilization(now pcommon.Timestamp, cpuUtilization ucal.CPUUtilization) {}
 
 func getProcessName(processHandle, string) (string, error) {
-	return nil, nil
+	return "", nil
 }
 
 func getProcessExecutable(processHandle) (string, error) {
-	return nil, nil
+	return "", nil
 }
 
 func getProcessCommand(processHandle) (*commandMetadata, error) {

